app/grandchallenge/participants/views.py

```diff
@@ -115,3 +126,103 @@ def get_success_url(self):
             "participants:registration-list",
             kwargs={"challenge_short_name": self.object.challenge.short_name},
         )
+
+
+class RegistrationQuestionList(
+    LoginRequiredMixin, PermissionListMixin, ListView
+):
+    model = RegistrationQuestion
+    permission_required = "participants.view_registrationquestion"
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        queryset = queryset.filter(challenge=self.request.challenge)
+        return queryset
+
+    def dispatch(self, request, *args, **kwargs):
+
+        if not self.request.user.has_perm(
+            "challenges.change_challenge",
+            self.request.challenge,
+        ):
+            # View contains admin side-panel
+            raise PermissionDenied
+
+        return super().dispatch(request, *args, **kwargs)
+
+
+class RegistrationQuestionCreate(
+    LoginRequiredMixin,
+    SuccessMessageMixin,
+    ObjectPermissionRequiredMixin,
+    CreateView,
+):
+    model = RegistrationQuestion
+    form_class = RegistrationQuestionForm
+
+    success_message = "Question successfully created"
+
+    permission_required = "challenges.add_registration_question"
+
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+
+    def get_permission_object(self):
+        return self.request.challenge
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["challenge"] = self.request.challenge
+        return kwargs
+
+    def get_success_url(self):
+        return reverse(
+            "participants:registration-question-list",
+            kwargs={"challenge_short_name": self.object.challenge.short_name},
+        )
+
+
+class RegistrationQuestionUpdate(
+    LoginRequiredMixin,
+    SuccessMessageMixin,
+    ObjectPermissionRequiredMixin,
+    UpdateView,
+):
+    model = RegistrationQuestion
+    form_class = RegistrationQuestionForm
+
+    permission_required = "participants.change_registrationquestion"
+
+    raise_exception = True
+    login_url = reverse_lazy("account_login")
+    success_message = "Question successfully updated"
+
+    def get_success_url(self):
+        return reverse(
+            "participants:registration-question-list",
+            kwargs={"challenge_short_name": self.object.challenge.short_name},
+        )
+
+
+class RegistrationQuestionDelete(
+    LoginRequiredMixin,
+    SuccessMessageMixin,
+    ObjectPermissionRequiredMixin,
+    DeleteView,
+):
+    model = RegistrationQuestion
+    fields = ()
```

Comment:
Still unresolved.

Reply:
`ModelFormMixin` (from `UpdateView`/`CreateView`), which I now realize was because of a copy+paste error when rewriting the views. I had not given the delete the correct base View but it gave me an error. The `DeleteView` has `FormMixin` as a base which does not require the `fields` defined. Removed!